src/lib.rs

```diff
@@ -47,9 +53,9 @@ pub fn run_compiled(book: &Book, mem_size: usize) -> (Net, RunStats) {
 }
 
 pub fn run_book(mut book: DefinitionBook, mem_size: usize) -> anyhow::Result<(Term, DefNames, RunInfo)> {
-  let compiled = compile_book(&mut book)?;
+  let (compiled, id_map) = compile_book(&mut book)?;
```

Comment:
Since inside `compile_book` we only use the left-to-right direction of the bimap and on readback we only use the `right-to-left` direction, I think it would be better to use just a normal HashMap.
Inside `compile_book` it can be just `id_to_hvmc_name` and then before returning you create a new HashMap with keys and values swapped with `let hvmc_name_to_id = id_to_hvmc_name.into_iter().map(|(k, v)| (v, k)).collect::<HashMap<_>>();`

Reply:
Another possible approach is to add this BiMap to the DefinitionBook or to the DefNames type, where it could be together with the other id to name map, which may be slightly more organized.
You could then make it calculate the hvmc name automatically when adding a definition to the book and it becomes accessible everywhere else without having to add extra arguments to the functions.

Reply:
I don't think the second aproach would be much better as the two functions that would use one of this Hashmaps don't need the DefinitionBook or DefNames as argument. So it would just change one argument for the other. Unless that is a good enough reason, as it simplifies the api